src/pages/api/oauth/refresh-tokens.ts

```diff
@@ -0,0 +1,65 @@
+import { HttpStatus } from "@fwl/web";
+import { ObjectId } from "mongodb";
+import type { NextApiResponse } from "next";
+import type { Handler } from "next-iron-session";
+
+import { refreshTokensFlow } from "@lib/refreshTokensFlow";
+import type { ExtendedRequest } from "@src/@types/ExtendedRequest";
+import { config } from "@src/config";
+import { withAPIPageLogger } from "@src/middleware/withAPIPageLogger";
+import { withMongoDB } from "@src/middleware/withMongoDB";
+import withSession from "@src/middleware/withSession";
+import Sentry, { addRequestScopeToSentry } from "@src/utils/sentry";
+
+const handler: Handler = async (
+  request: ExtendedRequest,
+  response: NextApiResponse,
+): Promise<void> => {
+  addRequestScopeToSentry(request);
+
+  try {
+    if (request.method === "POST") {
+      const { refreshToken, redirectUrl, userDocumentId } = request.body;
+
+      if (userDocumentId && refreshToken) {
+        const { refresh_token, access_token } = await refreshTokensFlow(
+          refreshToken,
+        );
+
+        const { result } = await request.mongoDb.collection("users").updateOne(
+          { _id: new ObjectId(userDocumentId) },
+          {
+            $set: { accessToken: access_token, refreshToken: refresh_token },
+          },
+        );
+
+        if (result.n === 0) {
+          throw new Error("Mongo update failed");
+        }
+
+        response.writeHead(HttpStatus.TEMPORARY_REDIRECT, {
+          Location: `${config.connectDomain}/${redirectUrl}`,
+        });
```

Comment:
@Fenntasy This is useless, right?

Reply:
Well... it will work and make the redirect instead of returning the response, but yes, if you want to send data, don't add a redirect ^^'